### PR TITLE
RNS (RIF Name Service) resolver support 

### PR DESCRIPTION
**Pull request by adil651**
### CHANGELOG

RNS: Support for RIF Name Service added

### Dependencies

 none

### Requirements

If you have made **any** visual changes to the GUI. Make sure you have:

- [x] Tested on iOS device
- [x] Tested on Android device
- [x] Tested on small-screen device (iPod Touch)
- [x] Tested on large-screen device (tablet)

This PR is a new version from https://github.com/EdgeApp/edge-react-gui/pull/2505 but this time with a library that doesn't need web3 or big-integer libraries

### CODE REVIEW

**Comment by adil651** [comment]
@paullinator Hello, Please review this pull request. I have removed all dependencies.

File: src/components/modals/AddressModal.tsx [hunk 1]
```diff
@@ -177,13 +181,93 @@ export class AddressModalComponent extends React.Component<Props, State> {
...
+      case 'eos':
+        return 194
+      default:
+        return -1
+    }
+  }
+
+  fetchRnsAddress = async (domain: string): Promise<string> => {
+    try {
+      const pluginId = this.getRnsChainIdFromPluginId(this.props.coreWallet.currencyInfo.pluginId)
```
Comment (paullinator) [inline src/components/modals/AddressModal.tsx:218]: this should be `chainId`

File: src/components/modals/AddressModal.tsx [hunk 2]
```diff
@@ -177,13 +181,93 @@ export class AddressModalComponent extends React.Component<Props, State> {
...
+      if (pluginId === -1) {
+        throw new ResolutionError('UnspecifiedCurrency', { domain })
+      }
+      if (!this.rskProvider) {
```
Comment (paullinator) [inline src/components/modals/AddressModal.tsx:222]: use explicit boolean checks as we will be making eslint stricter. 

`if (this.provider == null) {`

**Review by paullinator — CHANGES_REQUESTED** [review]

File: src/components/modals/AddressModal.tsx (hunk 2 above)
Comment (adil651) [reply src/components/modals/AddressModal.tsx]: @paullinator Changes made

File: src/components/modals/AddressModal.tsx (hunk 1 above)
Comment (adil651) [reply src/components/modals/AddressModal.tsx]: @paullinator Thank you for the review. Changes made, let me know if something else needs improvement

**Review by paullinator — APPROVED** [review]

**Comment by paullinator** [comment]
Replaced with rebased PR https://github.com/EdgeApp/edge-react-gui/pull/4138

**Pull request closed without merging**